src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -272,7 +272,8 @@ internal bool IsDerivedFrom(TypeSymbol type, bool ignoreDynamic, ref HashSet<Dia
             var t = this.BaseTypeWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics);
             while ((object)t != null)
             {
-                if (type.Equals(t, ignoreDynamic: ignoreDynamic))
+                var comparison = ignoreDynamicAndTupleNames ? (TypeCompareKind.IgnoreDynamicAndTupleNames) : TypeCompareKind.ConsiderEverything;
```

Comment:
Sorry about that. I did a search and replace. I should have reviewed the diff before pushing it for review.